src/main/java/uk/gov/justice/digital/delius/service/CustodyService.java

```diff
@@ -341,19 +347,11 @@ private void saveCustodyStatusChangeCustodyHistoryEvent(final Offender offender,
     }
 
     private Either<PrisonLocationUpdateError, Offender> findByNomsNumber(final String nomsNumber) {
-        final Supplier<Either<PrisonLocationUpdateError, Offender>> notFoundError = () -> Either
-                .left(PrisonLocationUpdateError.offenderNotFound(nomsNumber));
-
-        final Function<DuplicateOffenderException, Either<PrisonLocationUpdateError, Offender>> multipleOffenderError = duplicateError -> Either
-                .left(PrisonLocationUpdateError.multipleOffenders(duplicateError));
-
-        final Function<Optional<Offender>, Either<PrisonLocationUpdateError, Offender>> offenderOrNotFoundError = maybeOffender -> maybeOffender
-                .map((Function<Offender, Either<PrisonLocationUpdateError, Offender>>) Either::right)
-                .orElseGet(notFoundError);
-
         return offenderRepository
-                .findMostLikelyByNomsNumber(nomsNumber)
-                .fold(multipleOffenderError, offenderOrNotFoundError);
+            .findMostLikelyByNomsNumber(nomsNumber)
+            .mapLeft(PrisonLocationUpdateError::multipleOffenders)
```

Comment:
found previous version hard to understand with all the types everywhere, hopefully this version is clearer as to what is going on and do like the `mapLeft`

Reply:
Very nice.